pytest_unordered.py

```diff
@@ -1,20 +1,27 @@
 from typing import Generator
-from typing import Iterable
+from typing import Iterable, Mapping
 from typing import Sized
 
 from _pytest._io.saferepr import saferepr
 from _pytest.assertion.util import _compare_eq_any
 
 
 class UnorderedList(list):
-    def __init__(self, expected: Iterable):
+    def __init__(self, expected: Iterable, check_type: bool=True):
         if not isinstance(expected, Iterable):
             raise TypeError(
                 "cannot make unordered comparisons to non-iterable: {!r}".format(expected)
             )
+        if isinstance(expected, Mapping):
+            raise TypeError(
+                "cannot make unordered comparisons to mapping: {!r}".format(expected)
+            )
         super().__init__(expected)
+        self._expected_type = type(expected) if check_type else None
 
     def __eq__(self, actual: Iterable) -> bool:
+        if self._expected_type is not None and self._expected_type != type(actual):
+            return False
         if not isinstance(actual, Iterable):
             return self.copy() == actual
```

Comment:
After adding the type check, the condition `isinstance(actual, Iterable)` is always `False`.

Reply:
something like this visits those lines:

        assert unordered(1, 2, 3) == 3

Reply:
see PR #4